tests/system_tests/test_basic.py

```diff
@@ -50,19 +49,10 @@ def run_test_arguments(self, debug_info, expected_args):
     def run_test_termination(self, debug_info):
         with self.start_debugging(debug_info) as dbg:
             session = dbg.session
-
-            exited = session.get_awaiter_for_event('exited')
-            terminated = session.get_awaiter_for_event('terminated')
-
-            (_, req_launch, _, _, _, _
-             ) = lifecycle_handshake(dbg.session, debug_info.starttype,
-                                     threads=True)
-
-            Awaitable.wait_all(req_launch,
-                               session.get_awaiter_for_event('thread'))
-            disconnect = session.send_request('disconnect')
-
-            Awaitable.wait_all(exited, terminated, disconnect)
+            with self.wait_for_exit(session):
+                lifecycle_handshake(session, debug_info.starttype,
```

Comment:
Why are we not waiting for `exited`, `terminated`?

Reply:
That's what `self.wait_for_exit()` does.